### DISCUSSION TITLE
ILSpy vs competitors - ILSpy recognizes EXE, others dont?

### DISCUSSION THREAD

**Discussion by nullerror** [icsharpcode/ILSpy#2303] [General] [answered]
This is more of question than an issue or bug. I have an EXE that *only* ILSpy will recognize as a .NET app. DnSpy, Telerick, DotNetPeek, .Net Reflector, etc.. don't even recognize it. They just display the PE data and nothing else. 

It's an EXE that has 5 .NETCoreApps inside of it. What would you call this and why do the other programs not deal with it?

Thanks

**Comment by christophwille** [comment, marked as answer]
We support ReadyToRun (R2R) as well as .NET bundles. One of those (without seeing your program) is the most likely cause (bundles are only supported in v7p1+)

**Reply by siegfriedpammer** [reply]
From your description it seems this is a .NET bundle, yes, those are new in .NET Core 3.x/.NET 5.0 and support was added in ILSpy v7.0. As for the question why the other decompilers are not able to decompile them... no idea...

**Comment by nullerror** [comment]
Ah, that is great to know. I appreciate both of the quick replies. I just tried loading it up in version 6.2.1 and it doesn't work, so it looks like you guys are ahead of the game. Awesome job!
